src/app/pages/system/enclosure/components/enclosure-header/view-elements-menu/view-elements-menu.component.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  ChangeDetectionStrategy, Component, computed, input,
+} from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { EnclosureElementType, enclosureElementTypeLabels } from 'app/enums/enclosure-slot-status.enum';
+import { DashboardEnclosure } from 'app/interfaces/enclosure.interface';
+
+export interface ViewOption {
+  href: string;
+  label: string;
+}
+
+@Component({
+  selector: 'ix-view-elements-menu',
+  templateUrl: './view-elements-menu.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ViewElementsMenuComponent {
+  readonly enclosure = input.required<DashboardEnclosure>();
+
+  readonly views = computed<ViewOption[]>(() => {
+    const supportedViews = [
+      EnclosureElementType.ArrayDeviceSlot,
+      EnclosureElementType.TemperatureSensors,
+      EnclosureElementType.Cooling,
+      EnclosureElementType.VoltageSensor,
+      EnclosureElementType.PowerSupply,
+      EnclosureElementType.SasConnector,
+      EnclosureElementType.EnclosureServicesControllerElectronics,
+    ] as const;
+
+    return supportedViews
+      .filter((view) => this.enclosure().elements[view])
```

Comment:
Should declare a constant at the start and save the value and reuse it so it's clear this is a dependency for the computed signal.